src/System.Text.RegularExpressions/src/System/Buffers/MemoryOrPinnedSpan.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Buffers
+{
+    internal unsafe readonly struct MemoryOrPinnedSpan<T>
+    {
+        private readonly ReadOnlyMemory<T> _memory;
+        public readonly int Length;
+        private readonly char* _pinnedSpan;
+
+        public MemoryOrPinnedSpan(ReadOnlyMemory<T> memory)
+        {
+            _memory = memory;
+            Length = memory.Length;
+            _pinnedSpan = null;
+        }
+
+        public MemoryOrPinnedSpan(char* pinnedSpan, int length)
+        {
+            _pinnedSpan = pinnedSpan;
```

Comment:
Tiny nit, it is easier to read if initialization order is consistent and matches field order.

Reply:
I usually sort instance fields by their protection member. I switched the ordering now, I hope that's fine.

Reply:
The mix of public and private fields looks pretty weird to me.

Reply:
that's already changed with the latest commit.

Reply:
My comment was for what you have in the latest commit. I would write it like this:

```
private readonly ReadOnlyMemory<T> _memory; 

private readonly char* _pinnedSpan;
private readonly int _length; 
```

Reply:
ok cool, but I need to access the length from outside.

Reply:
Argh, hit enter to soon. I need to access the length from the MatchCollection that's why I made it public here.

Reply:
So make a property for it.

Reply:
Usually I'm fine with just exposing the field, if it's readonly, especially if the surrounding class is internal. But as we shouldn't spend too much time in discussing such a minor implementation detail I'll change it to a prop.